darts/models/forecasting/nbeats.py

```diff
@@ -108,13 +123,14 @@ def __init__(self,
         self.linear_layer_stack_list += [nn.Linear(layer_width, layer_width) for _ in range(num_layers - 1)]
         self.fc_stack = nn.ModuleList(self.linear_layer_stack_list)
 
-        # fully connected layer producing forecast/backcast expansion coeffcients (waveform generator parameters)
+        # Fully connected layer producing forecast/backcast expansion coeffcients (waveform generator parameters).
+        # The coefficients are emitted for each parameter of the likelihood.
         if g_type == _GType.SEASONALITY:
             self.backcast_linear_layer = nn.Linear(layer_width, 2 * int(input_chunk_length / 2 - 1) + 1)
-            self.forecast_linear_layer = nn.Linear(layer_width, 2 * int(target_length / 2 - 1) + 1)
+            self.forecast_linear_layer = nn.Linear(layer_width, nr_params * (2 * int(target_length / 2 - 1) + 1))
         else:
             self.backcast_linear_layer = nn.Linear(layer_width, expansion_coefficient_dim)
-            self.forecast_linear_layer = nn.Linear(layer_width, expansion_coefficient_dim)
+            self.forecast_linear_layer = nn.Linear(layer_width, nr_params * expansion_coefficient_dim)
```

Comment:
I think this is good now, we also have a nice consistency in how we create multiple likelihood parameters between generic and interpretable architectures.